pkg/artipie/artipie.go

```diff
@@ -0,0 +1,74 @@
+// Package artipie provides API for managing artipie server.
+package artipie
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// A Client manages communication with the Artipie API.
+type Client struct {
+	client    *http.Client
+	Endpoint  *url.URL
+	UserAgent string
+	Auth      Auth
+}
+
+// NewClient returns a new Artipie API client. If a nil httpClient is
+// provided, a new http.Client will be used.
+// auth should implement Auth interface and be able to set authentification headers to request.
+// auth could be nil, in that case http.client should provide authentification.
+func NewClient(client *http.Client, endpoint string, auth Auth) (*Client, error) {
+	if client == nil {
+		client = &http.Client{}
+	}
+	if endpoint == "" {
+		return nil, errors.New("Endpoint is not specified")
+	}
+	ep, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{client: client, Endpoint: ep, Auth: auth}, nil
+}
+
+// NewRequest creates an API request.
+// If specified, the value pointed to by body is JSON encoded and included as the
+// request body.
+func (c *Client) NewRequest(method, path string, body interface{}) (*http.Request, error) {
```

Comment:
Body encoding is usually a higher-level operation than client-serevr communication, I'd suggest accepting here raw body (e.g. `io.Reader`) and use it as request body. We may want to send non-JSON requests body for API endpoint

Reply:
I think it will be usefull to have method that creates requests and marshals JSON struct.
It will be the most common usage of api client.
So I propose to use NewJSONRequest as a wrapper for NewRequest.

Reply:
@g4s8 please, review!